exporter/otlphttpexporter/config.go

```diff
@@ -1,73 +1,144 @@
-// Copyright The OpenTelemetry Authors
-// SPDX-License-Identifier: Apache-2.0
+// Code generated by github.com/atombender/go-jsonschema, DO NOT EDIT.
 
-package otlphttpexporter // import "go.opentelemetry.io/collector/exporter/otlphttpexporter"
+package otlphttpexporter
 
-import (
-	"encoding"
-	"errors"
-	"fmt"
+import "encoding/json"
+import "fmt"
+import confighttp "go.opentelemetry.io/collector/config/confighttp"
+import configretry "go.opentelemetry.io/collector/config/configretry"
+import exporterhelper "go.opentelemetry.io/collector/exporter/exporterhelper"
+import "reflect"
 
-	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/config/confighttp"
-	"go.opentelemetry.io/collector/config/configretry"
-	"go.opentelemetry.io/collector/exporter/exporterhelper"
-)
+// Configuration parameters for the OTLP HTTP exporter.
+type Config struct {
+	// Auth corresponds to the JSON schema field "auth".
+	Auth *confighttp.Auth `mapstructure:"auth,omitempty"`
 
-// EncodingType defines the type for content encoding
-type EncodingType string
+	// Compression corresponds to the JSON schema field "compression".
+	Compression *confighttp.Compression `mapstructure:"compression,omitempty"`
 
-const (
-	EncodingProto EncodingType = "proto"
-	EncodingJSON  EncodingType = "json"
-)
+	// Cookies corresponds to the JSON schema field "cookies".
+	Cookies *confighttp.Cookies `mapstructure:"cookies,omitempty"`
 
-var _ encoding.TextUnmarshaler = (*EncodingType)(nil)
+	// DisableKeepAlive corresponds to the JSON schema field "disable_keep_alive".
+	DisableKeepAlive *confighttp.DisableKeepAlive `mapstructure:"disable_keep_alive,omitempty"`
 
-// UnmarshalText unmarshalls text to an EncodingType.
-func (e *EncodingType) UnmarshalText(text []byte) error {
-	if e == nil {
-		return errors.New("cannot unmarshal to a nil *EncodingType")
-	}
+	// Encoding corresponds to the JSON schema field "encoding".
+	Encoding ConfigEncoding `mapstructure:"encoding,omitempty"`
 
-	str := string(text)
-	switch str {
-	case string(EncodingProto):
-		*e = EncodingProto
-	case string(EncodingJSON):
-		*e = EncodingJSON
-	default:
-		return fmt.Errorf("invalid encoding type: %s", str)
-	}
+	// Endpoint corresponds to the JSON schema field "endpoint".
+	Endpoint confighttp.Endpoint `mapstructure:"endpoint,omitempty"`
 
-	return nil
-}
+	// Headers corresponds to the JSON schema field "headers".
+	Headers confighttp.Headers `mapstructure:"headers,omitempty"`
 
-// Config defines configuration for OTLP/HTTP exporter.
-type Config struct {
-	confighttp.ClientConfig `mapstructure:",squash"`     // squash ensures fields are correctly decoded in embedded struct.
```

Comment:
Unfortunately, I can't find a good way to schematise squashed fields. **This is my main blocker right now.**

I was hoping that we can use something like [anyOf](https://json-schema.org/understanding-json-schema/reference/combining#anyOf), but judging by some [examples](https://github.com/omissis/go-jsonschema/pull/99/files), it also looks awkward.

For now, I just listed all the squashed properties explicitly in `metadata.yaml`. Does anyone know of a better way? I'm not sure what folks working with Go code normally do in such situations.

Reply:
Hi, @atoulme! Good news! 🥳  I was able to find a way to create squashed structs with go-jsonschema. I will update this PR next week, once I polish my code a bit. The new generated code will look a lot more like the current OTel code.

Reply:
🚀